src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -114,42 +166,81 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
 
             var templateContext = new TemplateContext
             {
-                OriginalTemplate = JObject.Parse(template),
+                OriginalTemplate = JObject.Parse(populatedTemplate),
                 ExpandedTemplate = templatejObject,
-                IsMainTemplate = true,
+                IsMainTemplate = parentContext.ParentContext == null,
```

Comment:
The logic here is a little weird, since this design means the top-level context will still have a fake "parent".  To make it a little clearer what's going on, it might help to check for `parentContext.OriginalTemplate == null` instead, denoting the passed context doesn't actually represent a processed template.  A comment would help here too.